src/assemblies.ts

```diff
@@ -1,18 +1,14 @@
 import * as path from 'path';
-import * as spec from '@jsii/spec';
+import { Assembly, Type, writeAssembly, compressedAssemblyExists } from '@jsii/spec';
 import * as fs from 'fs-extra';
 import { FIXTURE_NAME } from './generate-missing-examples';
 
 /**
  * Replaces the file where the original assembly file *should* be found with a new assembly file.
- * Recalculates the fingerprint of the assembly to avoid tampering detection.
+ * Detects whether or not there is a compressed assembly, and if there is, compresses the new assembly also.
  */
-export async function replaceAssembly(assembly: spec.Assembly, directory: string): Promise<void> {
-  const fileName = path.join(directory, '.jsii');
-  await fs.writeJson(fileName, _fingerprint(assembly), {
-    encoding: 'utf8',
-    spaces: 2,
-  });
+export function replaceAssembly(assembly: Assembly, directory: string) {
```

Comment:
This entire function is a direct copy of:

https://github.com/aws/jsii/blob/main/packages/jsii-rosetta/lib/jsii/assemblies.ts#L244-L246

**Why is it a duplicate?**

The linked function is inside `jsii-rosetta` and is not exported. The decision to duplicate this function rather than expose it in `jsii-rosetta`was made because it does not pertain to `jsii-rosetta` and is more of an implementation detail. 

Also, it also allows us to invalidate the fingerprint, replacing it with `********` instead of a new fingerprint. But we do not actually use the fingerprint, given this quote from line 29 in this file: "@rmuller says fingerprinting is useless, as we do not actually check if an assembly is changed." 

I then remembered that I was supposed to do the same in jsii, so I tracked that down and well, I never actually got it merged: https://github.com/aws/jsii/pull/3146/files. That tells me that the behavior here was never meant to be different from the behavior in jsii, so this is not a real reason to be duplicating the function. I can (and should) reopen the jsii PR to add this behavior to jsii.

**What am I going to do after this PR is merged?**

Given that this duplication has caused me plenty of stress, and that the next time we modify `replaceAssembly` the same thing will happen, I think this is no longer acceptable. I will open a PR in jsii to move `replaceAssembly` to `@jsii/spec` and export it there, and when jsii is released, I will update this to use that.

This current change is a stopgap to fix the current issue; exporting `replaceAssembly` will make sure it never happens again.

Reply:
You plan to completely remove the `replaceAssembly` function from this package, right? And just reference the one that is exported from `@jsii/spec`?

Reply:
yes exactly